### PR TITLE
sirula: init at unstable-2021-07-11

### PR DESCRIPTION
**Pull request by twitchyliquid64**
This PR packages sirula, a simple app launcher for wayland.

###### Motivation for this change

To package [sirula](https://github.com/DorianRudolph/sirula)

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions - Debian Buster
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132241` at d4719df4 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>sirula</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/tools/wayland/sirula/default.nix:29:5:
```
   |
29 |     license = with licenses; [ gpl3 ];
   |     ^
```
</li>
  </ul>
</details>

**Comment by twitchyliquid64** [comment]
Thanks for the review! I've updated as `gpl3Plus` as the license blurb specifies "... or (at your option) any later version."

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -11006,6 +11006,12 @@
       fingerprint = "E631 8869 586F 99B4 F6E6  D785 5942 58F0 389D 2802";
     }];
   };
+  twitchyliquid64 = {
+    name = "Tom";
+    email = "twitchyliquid64@ciphersink.net";
+    github = "twitchyliquid64";
+    githubId = 6328589;
+  };
```
Comment (legendofmiracles) [inline maintainers/maintainer-list.nix:11014]: Separate commit: `maintainers: add twitchyliquid64`

File: pkgs/tools/wayland/sirula/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, gtk3
+, gtk-layer-shell
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "sirula";
+  version = "574725bc307fc704c42380cd0fa50b0b80c4764d";
```
Comment (legendofmiracles) [inline pkgs/tools/wayland/sirula/default.nix:11]: Version should be the date with a `unstable-` added before it (https://nixos.org/manual/nixpkgs/stable/#sec-package-naming)
> If a package is not a release but a commit from a repository, then the version part of the name must be the date of that (fetched) commit. The date must be in "YYYY-MM-DD" format. Also append "unstable" to the name - e.g., "pkgname-unstable-2014-09-23".

**Comment by twitchyliquid64** [comment]
Done. Thank you @legendofmiracles :]

**Review by legendofmiracles — APPROVED** [review]

**Comment by tomberek** [comment]
Result of `nixpkgs-review pr 132241` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>sirula</li>
  </ul>
</details>

**Pull request merged**